### PR TITLE
getsubstate attempt #1

### PR DESCRIPTION
**Pull request by JohnnyJumper**
* Getting substate by looking for vname === variableName in args.

Question:

Imagine I have such state:

> [
  {
    "vname": "_balance",
    "type": "Uint128",
    "value": "0"
  },
  {
    "vname": "welcome_msg",
    "type": "String",
    "value": "Hello World"
  },
  {
    "vname": "test",
    "type": "Map",
    "value": {
      "addr1": {
        "vname": "secondTest",
        "type": "Map",
        "value": {
          "addr2": {
            "vname": "answer",
            "type": "string",
            "value": "this should returned"
          }
        }
      }
    }
  }
]

How would I use getSubContract method if I want to get addr2? it should be called as ("test", ["addr1", "addr2"] ?

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -18,7 +18,6 @@ const bodyParser = require('body-parser');
 const cors = require('cors');
 const express = require('express');
 const fs = require('fs');
-const rimraf = require('rimraf');
```
Comment (JohnnyJumper) [inline src/app.js]: Wasn't using so I removed this

File: src/logic.js [hunk 2]
```diff
@@ -436,6 +438,16 @@ module.exports = {
     if (fileType === 'code') {
       return { code: responseData };
     }
+
+    if (isSubState) {
+      logVerbose(logLabel, `Retrieving subState ${data[1]} from ${filePath}`);
+      if (data.length < 2) { return JSON.parse(responseData); }
+      const variableName = data[1];
+      const subResponseData = JSON.parse(responseData).find(v => v.vname === variableName);
+      return { [variableName]: subResponseData };
```
Comment (bogdan) [inline src/logic.js]: So what about the second level when you query particular map value if the field is a map?

**Comment by bogdan** [comment]
Please add a test for this functionality.

File: src/logic.js (hunk 2 above)
Comment (JohnnyJumper) [reply src/logic.js]: At the moment nothing happens, how should I approached this task? I understand that I need to be able to go as deep as needed to get the state, but which value should I look for vname or the key inside the first level value ? 

And what supposed to be querying vnames or keys? I got confused on that one.

Comment (JohnnyJumper) [reply src/logic.js]: And if a key hasn't been found somewhere in the middle of process what error should I throw?

Comment (bogdan) [reply src/logic.js]: For the errors cases: use the live blockchain calls to see how live node reacts on non-exitsing field. 

See how it works in namicorn live blockchain calls and reimplement that behavior. 
Their RPC docs says there are 3 arguments - you've only implemented two. 
There is no doc for 3rd argument, but you can see how it works in namicorn.

**Comment by bogdan** [comment]
Did you delete the `package-lock.json` file? That is bad idea for a repo that is not made by us. You can do that only in namicorn.

**Pull request closed without merging**